Support -fvisibility=hidden

Compiling with -fvisibilty=hidden worked a few releases agao but the use of libIcdLog.so causes these errors:

```
[100%] Linking C executable ../../icd_loader_test
CMakeFiles/icd_loader_test.dir/main.c.o: In function `main':
main.c:(.text.startup+0x2): undefined reference to `test_icd_initialize_app_log'
main.c:(.text.startup+0x7): undefined reference to `test_icd_initialize_stub_log'
main.c:(.text.startup+0x52): undefined reference to `test_icd_close_app_log'
main.c:(.text.startup+0x57): undefined reference to `test_icd_close_stub_log'
CMakeFiles/icd_loader_test.dir/test_kernel.c.o: In function `test_clRetainKernel':
test_kernel.c:(.text+0x15): undefined reference to `test_icd_app_log'
test_kernel.c:(.text+0x31): undefined reference to `test_icd_app_log'
CMakeFiles/icd_loader_test.dir/test_kernel.c.o: In function `test_clSetKernelArg':
test_kernel.c:(.text+0x60): undefined reference to `test_icd_app_log'
test_kernel.c:(.text+0x87): undefined reference to `test_icd_app_log'
CMakeFiles/icd_loader_test.dir/test_kernel.c.o: In function `test_clGetKernelInfo':
test_kernel.c:(.text+0xb4): undefined reference to `test_icd_app_log'
CMakeFiles/icd_loader_test.dir/test_kernel.c.o:test_kernel.c:(.text+0xdf): more undefined references to `test_icd_app_log' follow
CMakeFiles/icd_loader_test.dir/icd_test_match.c.o: In function `test_icd_match':
icd_test_match.c:(.text+0x5): undefined reference to `test_icd_get_app_log'
icd_test_match.c:(.text+0x12): undefined reference to `test_icd_get_stub_log'
CMakeFiles/icd_loader_test.dir/test_program_objects.c.o: In function `test_clRetainProgram':
test_program_objects.c:(.text+0x15): undefined reference to `test_icd_app_log'
test_program_objects.c:(.text+0x31): undefined reference to `test_icd_app_log'
CMakeFiles/icd_loader_test.dir/test_program_objects.c.o: In function `test_clBuildProgram':
test_program_objects.c:(.text+0x6e): undefined reference to `test_icd_app_log'
test_program_objects.c:(.text+0xa0): undefined reference to `test_icd_app_log'
CMakeFiles/icd_loader_test.dir/test_program_objects.c.o: In function `test_clCompileProgram':
test_program_objects.c:(.text+0xe4): undefined reference to `test_icd_app_log'
CMakeFiles/icd_loader_test.dir/test_program_objects.c.o:test_program_objects.c:(.text+0x127): more undefined references to `test_icd_app_log' follow
collect2: error: ld returned 1 exit status
gmake[2]: *** [icd_loader_test] Error 1
gmake[1]: *** [test/loader_test/CMakeFiles/icd_loader_test.dir/all] Error 2
gmake: *** [all] Error 2
```

Can we update test/inc/platform/icd_test_log.h to handle this?
```
#if defined (_WIN32)
#define DllExport   __declspec( dllexport ) 
#else
#define DllExport  __attribute__((visibility("default")))
#endif
```
Possibly check for __GNUC__ or __clang__?  I can make a pull request if this would be welcome.

--- a/test/inc/platform/icd_test_log.h
+++ b/test/inc/platform/icd_test_log.h
@@ -1,10 +1,18 @@
 #ifndef _ICD_TEST_LOG_H_
 #define _ICD_TEST_LOG_H_
 
-#if defined (_WIN32)
-#define DllExport   __declspec( dllexport ) 
+#if defined(_WIN32) || defined(__CYGWIN__)
+    #define DllExport __declspec(dllexport)
 #else
-#define DllExport
+    #ifndef __has_attribute
+        #define __has_attribute(x) 0  // Compatibility with non-clang compilers.
+    #endif
+    #if (defined(__GNUC__) && (__GNUC__ >= 4)) ||\
+        (defined(__clang__) && __has_attribute(visibility))
+        #define DllExport __attribute__((visibility("default")))
+    #else
+        #define DllExport
+    #endif
 #endif
 
 DllExport int test_icd_initialize_app_log(void);

